src/pages/api/auth-connect/mark-identity-as-primary.ts

```diff
@@ -42,27 +56,38 @@ const handler: Handler = async (request, response) => {
           true,
         );
 
-        return markIdentityAsPrimary(
-          config.managementCredentials,
-          identityId,
-        ).then(() => {
-          span.setDisclosedAttribute("is Identity marked as primary", true);
-
-          response.statusCode = HttpStatus.OK;
-          response.setHeader("Content-type", "application/json");
-          response.end();
-          return;
-        });
+        return markIdentityAsPrimary(config.managementCredentials, identityId)
+          .then(() => {
+            span.setDisclosedAttribute("is Identity marked as primary", true);
+
+            response.statusCode = HttpStatus.OK;
+            response.setHeader("Content-type", "application/json");
+            response.end();
+            return;
+          })
+          .catch((error) => {
+            span.setDisclosedAttribute("is Identity marked as primary", false);
+
+            if (error instanceof GraphqlErrors) {
+              span.setDisclosedAttribute("Identity not found", error.message);
+              throw webErrorFactory(webErrors.identityNotFound);
+            }
+
+            if (error instanceof ConnectUnreachableError) {
+              span.setDisclosedAttribute("Connect unreachable", error.message);
+              throw webErrorFactory(webErrors.connectUnreachable);
+            }
+
+            throw error;
+          });
       }
 
       span.setDisclosedAttribute(
         "is identity marked as primary authorized",
         false,
       );
 
-      response.statusCode = HttpStatus.BAD_REQUEST;
-      response.end();
-      return;
+      throw webErrorFactory(webErrors.badRequest);
```

Comment:
What would you think about reversing this `if`?
Meaning having the bad path happening before the happy path:

```
if (!isAuthorized) {
  span.setDisclosedAttribute(
    "is identity marked as primary authorized",
    false,
  );
  throw webErrorFactory(webErrors.badRequest);
}

// rest of the code
```

I feel this would be more readable, what do you think? (ping @Fuzznimp )

Reply:
I'm all for it. I've started to do this, and in most cases it's way more clean.

Reply:
Good for me too 👍